pandas/core/frame.py

```diff
@@ -8539,51 +8539,42 @@ def blk_func(values):
                 out[:] = coerce_to_dtypes(out.values, df.dtypes)
             return out
 
-        if not self._is_homogeneous_type:
-            # try to avoid self.values call
-
-            if filter_type is None and axis == 0 and len(self) > 0:
-                # operate column-wise
-
-                # numeric_only must be None here, as other cases caught above
-                # require len(self) > 0 bc frame_apply messes up empty prod/sum
-
-                # this can end up with a non-reduction
-                # but not always. if the types are mixed
-                # with datelike then need to make sure a series
-
-                # we only end up here if we have not specified
-                # numeric_only and yet we have tried a
-                # column-by-column reduction, where we have mixed type.
-                # So let's just do what we can
-                from pandas.core.apply import frame_apply
-
-                opa = frame_apply(
-                    self, func=f, result_type="expand", ignore_failures=True
-                )
-                result = opa.get_result()
-                if result.ndim == self.ndim:
-                    result = result.iloc[0].rename(None)
-                return result
-
         if numeric_only is None:
             data = self
             values = data.values
-
             try:
                 result = f(values)
 
             except TypeError:
                 # e.g. in nanops trying to convert strs to float
 
+                # try by-column first
```

Comment:
why does this need to be moved?  if it is moved, then the "try by-column first" comment is no longer accurate

Reply:
> the "try by-column first" comment is no longer accurate

This comment was there originally before the move in https://github.com/pandas-dev/pandas/pull/32950/files.

> why does this need to be moved?

This PR reverts a change that caused a regression. The PR that caused the regression is labelled as a clean. This PR is in response to https://github.com/pandas-dev/pandas/issues/34730#issuecomment-653208905.